spark/src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala

```diff
@@ -1848,6 +1848,13 @@ trait OptimisticTransactionImpl extends TransactionalWrite
     val info = currentTransactionInfo.commitInfo
       .map(_.copy(readVersion = None, isolationLevel = None)).orNull
     setNeedsCheckpoint(attemptVersion, postCommitSnapshot)
+    val doCollectCommitStats =
+      needsCheckpoint || spark.sessionState.conf.getConf(DeltaSQLConf.DELTA_COLLECT_COMMIT_STATS)
```

Comment:
nit: Given that now we sometimes collect commit stats anyway, should we rename this as `DELTA_FORCE_ALL_COMMIT_STATS` (default false)?

Reply:
Should I also change to `buildConf("commitStats.force")` instead of `commitStats.collect`?

Reply:
Seems reasonable, if you didn't already do it?

Reply:
Yes I did it already :)